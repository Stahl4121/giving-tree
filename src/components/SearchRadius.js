import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";


// 1) Radius should be set to none and disabled when nothing is in search.
// 2) Radius should be enabled and default to 25mi when location is chosen.
// 3) (1) and (2) should properly switch between each other.

const useStyles = makeStyles(theme => ({
  formControl: {
    minWidth: 120
  },
  selectEmpty: {
    marginTop: theme.spacing(2)
  }
}));

export default function SearchRadius(props) {
  const classes = useStyles();
  const [radius, setRadius] = React.useState(-1);

  const inputLabel = React.useRef("None");
  const [labelWidth, setLabelWidth] = React.useState(0);
  React.useEffect(() => {
    setLabelWidth(inputLabel.current.offsetWidth);
  }, []);

  const handleChange = event => {
    setRadius(event.target.value);
    props.parentCallback(event.target.value);
  };

  React.useEffect(() => {
    if (props.loc && radius == -1) {
      setRadius(25)
      props.parentCallback(25);
    }
    if (!props.loc) {
      setRadius(-1)
      props.parentCallback(-1);
    }
  }, [props.loc])

  return (
    <div>
      <FormControl fullWidth variant="outlined" className={classes.formControl}>
        <InputLabel ref={inputLabel} id="select-radius-ref">
          Radius (mi)
        </InputLabel>
        <Select
          labelId="select-radius-label"
          id="select-radius"
          value={radius}
          onChange={handleChange}
          labelWidth={labelWidth}
          fullWidth
        >
          <MenuItem value={-1}>
            <em>None</em>
          </MenuItem>
          <MenuItem value={10}>10</MenuItem>
          <MenuItem value={25}>25</MenuItem>
          <MenuItem value={50}>50</MenuItem>
          <MenuItem value={100}>100</MenuItem>
        </Select>
      </FormControl>
    </div>
  );
}
