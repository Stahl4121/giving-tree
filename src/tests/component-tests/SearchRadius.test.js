import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SearchRadius from "../../components/SearchRadius.js";

describe('Component: SearchRadius', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SearchRadius parentCallback={onSelect} />)
    expect(wrap).toMatchSnapshot();
  });

  it('Select 10 miles', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SearchRadius parentCallback={onSelect} />)
    const select = wrap.find('#select-radius');
    console.log("//////////////////////////////////////////////////");
    console.log("select: ", select);
    console.log("//////////////////////////////////////////////////");
    //wrapper.find(Component).at(index).simulate('click');
    //select.
    select.simulate('change', { target: { value: '10'} });
    expect(onSelect.mock.calls.length).toBe(1);
    expect(onSelect.mock.calls[0][0].toBe('10'));
    //select.simulate('keydown', { keyCode: 13 });
    //expect(wrap.find('SearchRadius').radius).toEqual(10);//('radius')).toEqual(10)
  });

  it('Select 25 miles', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SearchRadius parentCallback={onSelect} />)
    const select = wrap.find('#select-radius').hostNodes();
    select.simulate('change', { target: { value: '25'} });
    //select.simulate('keydown', { keyCode: 13 });
    expect(wrap.find('SearchRadius').radius).toEqual(25)
  });

  it('Select 50 miles', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SearchRadius parentCallback={onSelect} />)
    const select = wrap.find('#select-radius').hostNodes();
    select.simulate('change', { target: { value: '50'} });
    //select.simulate('keydown', { keyCode: 13 });
    expect(wrap.find('SearchRadius').radius).toEqual(50)
  });

  it('Select 100 miles', () => {
    const onSelect = jest.fn();
    const wrap = enzyme.mount(<SearchRadius parentCallback={onSelect} />)
    const select = wrap.find('#select-radius').hostNodes();
    select.simulate('change', { target: { value: '100'} });
    //select.simulate('keydown', { keyCode: 13 });
    expect(wrap.find('SearchRadius').radius).toEqual(100)
  });

})