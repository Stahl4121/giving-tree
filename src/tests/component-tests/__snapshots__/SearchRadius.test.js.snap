// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchRadius Renders correctly 1`] = `ReactWrapper {}`;
